jerry-core/api/jerry-snapshot.c

```diff
@@ -24,15 +24,50 @@
 #include "lit-char-helpers.h"
 #include "re-compiler.h"
 
+#if defined JERRY_ENABLE_SNAPSHOT_SAVE || defined JERRY_ENABLE_SNAPSHOT_EXEC
+
+/**
+ * Get snapshot configuration flags.
+ *
+ * @return configuration flags
+ */
+static inline uint32_t __attr_always_inline___
+snapshot_get_configuration (void)
```

Comment:
It's somewhat counterintuitive to see that `snapshot_check_configuration` does something with a flag named `JERRY_SNAPSHOT_HAS_REGEX_LITERAL` but this `snapshot_get_configuration` does nothing with the same flag. Later I found that there is a fixup for the config flags in `jerry_parse_and_save_snapshot` like:
```c
  header.configuration = snapshot_get_configuration ();
#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
  if (globals.regex_found)
  {
    header.configuration |= JERRY_SNAPSHOT_HAS_REGEX_LITERAL;
  }
#endif /* CONFIG_DISABLE_REGEXP_BUILTIN */
```

One would think that config flag management would be the responsibility of these two functions (only). What about changing this function a bit like:

```c
static inline uint32_t __attr_always_inline___
snapshot_get_configuration (bool regex)
{
  return (0
#ifdef JERRY_CPOINTER_32_BIT
          | JERRY_SNAPSHOT_FOUR_BYTE_CPOINTER
#endif /* JERRY_CPOINTER_32_BIT */
#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
          | (regex ? JERRY_SNAPSHOT_HAS_REGEX_LITERAL : 0)
#endif /* CONFIG_DISABLE_REGEXP_BUILTIN */
          | 0);
} /* snapshot_get_configuration */
```

Then, we could simply have
```c
  header.configuration = snapshot_get_configuration (globals.regex_found);
```
at the call site in `jerry_parse_and_save_snapshot`.

Moreover, `return configuration == snapshot_get_configuration (false);` in `snapshot_check_configuration` below.

Reply:
I will not change this. The difference is that the 'regex' is a property and not a configuration.

Reply:
(Note: I don't want to use the `configuration_and_properties` because I feel it is too long)

Reply:
I'd like you to reconsider this. Even the proposed code calls all these flags configuration flags, there is no mentioning of properties:
```c
/**
 * Snapshot configuration flags.
 */
typedef enum
{
  /* 8 bits are reserved for dynamic features */
  JERRY_SNAPSHOT_HAS_REGEX_LITERAL = (1u << 0), /**< byte code has regex literal */
  /* 24 bits are reserved for compile time features */
  JERRY_SNAPSHOT_FOUR_BYTE_CPOINTER = (1u << 8) /**< compressed pointers are four byte long */
} jerry_snapshot_configuration_flags_t;
```
The only difference is between dynamic and compile time features.

Alternative suggestion:
```c
static inline uint32_t __attr_always_inline___
snapshot_get_configuration (uint32_t dynamic_feature_flags)
{
  return (0
#ifdef JERRY_CPOINTER_32_BIT
          | JERRY_SNAPSHOT_FOUR_BYTE_CPOINTER
#endif /* JERRY_CPOINTER_32_BIT */
#ifndef CONFIG_DISABLE_REGEXP_BUILTIN
          | (dynamic_feature_flags & JERRY_SNAPSHOT_HAS_REGEX_LITERAL)
#endif /* !CONFIG_DISABLE_REGEXP_BUILTIN */
          | 0);
} /* snapshot_get_configuration */
```

I do think that config flag management should not be scattered around the code base (should there come more flags in the future) but managed by a few functions.

Reply:
I just wanted to keep things simple. But lets call them global flags.